README.md

```diff
@@ -318,24 +341,60 @@ intervals.
 
 Fire DOM events.
 
+React attaches an event handler on the `document` and handles some DOM events via event delegation (events bubbling up from a `target` to an ancestor). Because of this, your `node` must be in the `document.body` for `fireEvent` to work with React. You can render into the document using the [renderIntoDocument](#renderintodocument) utility. This is an alternative to simulating Synthetic React Events via [Simulate](#simulate). The benefit of using `fireEvent` over `Simulate` is that you are testing real DOM events instead of Synthetic Events. This aligns better with [the Guiding Principles](#guiding-principles).
+
 ```javascript
-// <button>Submit</button>
-fireEvent(
-  getElementByText('Submit'),
-  new MouseEvent('click', {
-    bubbles: true,
-    cancelable: true,
-  }),
-)
+import { renderIntoDocument, clearDocument, render, fireEvent }
+
+// don't forget to clean up the document.body
+afterEach(clearDocument)
+
+test('clicks submit button', () => {
+  const spy = jest.fn();
+  const { unmount, getByText } = render(<button onClick={spy}>Submit</button>)
+
+  fireEvent(
+    getByText('Submit'),
+    new MouseEvent('click', {
+      bubbles: true, // click events must bubble for React to see it
+      cancelable: true,
+    })
+  )
+
+  // don't forget to unmount component so componentWillUnmount can clean up subscriptions
+  unmount();
+
+  expect(spy).toHaveBeenCalledTimes(1);
+});
 ```
 
-#### `fireEvent[eventName](node: HTMLElement, eventInit)`
+#### `fireEvent[eventName](node: HTMLElement, eventProperties: Object)`
 
-Convenience methods for firing DOM events. Look [here](./src/events.js) for full list.
+Convenience methods for firing DOM events. Check out
+[dom-testing-library/src/events.js](https://github.com/kentcdodds/dom-testing-library/blob/master/src/events.js)
+for a full list as well as default `eventProperties`.
 
 ```javascript
-// <button>Submit</button>
-fireEvent.click(getElementByText('Submit'))
+import { renderIntoDocument, clearDocument, render, fireEvent }
+
+// don't forget to clean up the document.body
+afterEach(clearDocument)
+
+test('right clicks submit button', () => {
+  const spy = jest.fn();
+  const onClick = e => e.button === 2 && spy();
+  const { unmount, getByText } = render(<button onClick={onClick}>Submit</button>)
+
+  // click will bubble for React to see it
+  const rightClick = {button: 2}
+  fireEvent.click(getElementByText('Submit'), rightClick)
+  // default `button` property for click events is set to `0` which is a left click.
+
+  // don't forget to unmount component so componentWillUnmount can clean up subscriptions
```

Comment:
ok we can do it in the library, but i think this is even more reason to have `cleanup` be in an `afterEach` hook so if a `componentWillUnmount` throws an error it will be associated with the correct test.

Reply:
done but kept `beforeEach` in examples

Reply:
> ok we can do it in the library, but i think this is even more reason to have cleanup be in an afterEach hook so if a componentWillUnmount throws an error it will be associated with the correct test.

This makes a lot of sense. Sorry, but could you change it back? 😅

Reply:
haha np, will change it back

Reply:
done